### PR TITLE
Make Helm-Broker more resilient

### PR DESCRIPTION
**Pull request by piotrmiskiewicz**
**Description**

Changes proposed in this pull request:

- Helm-Broker retries index.yaml reading.
- Helm-Broker does not fail if bundle is corrupted
- repository URL configured as an environment variable
- change `yBundle` to `bundle`

**Related issue(s)**

#592

### CODE REVIEW

File: docs/service-brokers/docs/011-configuration-helm-broker.md [hunk 1]
```diff
@@ -33,12 +33,33 @@ Follow these steps to change the configuration and make the Helm Broker fetch bu
...
 
   ```yaml
-  config:
     repository:
       baseURL: "http://custom.bundles-repository"
   ```
 
 3. Install Kyma on Minikube. See the **Local Kyma installation** document to learn how.
+
+### Configuring repository URL in the runtime
```
Comment (klaudiagrz) [inline docs/service-brokers/docs/011-configuration-helm-broker.md]: `### Configure the repository URL in the runtime`

File: docs/service-brokers/docs/011-configuration-helm-broker.md [hunk 2]
```diff
@@ -33,12 +33,33 @@ Follow these steps to change the configuration and make the Helm Broker fetch bu
...
+
+Follow these steps to change the repository URL:
+
+1. Configure Helm Broker:
```
Comment (klaudiagrz) [inline docs/service-brokers/docs/011-configuration-helm-broker.md]: `APP_REPOSITORY_BASE_URL="http://custom.bundles-repository/"` -> missing " at the end?

File: docs/service-brokers/docs/011-configuration-helm-broker.md [hunk 3]
```diff
@@ -33,12 +33,33 @@ Follow these steps to change the configuration and make the Helm Broker fetch bu
...
+
+ ```bash
+ kubectl set env -n kyma-system deployment/core-helm-broker -e APP_REPOSITORY_BASE_URL="http://custom.bundles-repository/
+ ```
+ 
+2. Wait for running Helm Broker using the following command:
```
Comment (klaudiagrz) [inline docs/service-brokers/docs/011-configuration-helm-broker.md]: `2. Wait for the Helm Broker to run using the following command:` ?

File: docs/service-brokers/docs/011-configuration-helm-broker.md [hunk 4]
```diff
@@ -33,12 +33,33 @@ Follow these steps to change the configuration and make the Helm Broker fetch bu
...
+
+ ```bash
+ kubectl get pod -n kyma-system -l app=core-helm-broker
+ ```
+
+3. Trigger Service Catalog synchronization:
```
Comment (klaudiagrz) [inline docs/service-brokers/docs/011-configuration-helm-broker.md]: `3. Trigger the Service Catalog synchronization:`

File: components/helm-broker/cmd/broker/main.go [hunk 5]
```diff
@@ -49,7 +52,14 @@ func main() {
 		ybundle.NewHTTPRepository(cfg.Repository),
 		ybLoader, sFact.Bundle(), sFact.Chart(), log,
 	)
-	err = cachePop.Init()
+	for i := 0; i < maxPopulatorRetries; i++ {
```
Comment (polskikiel) [inline components/helm-broker/cmd/broker/main.go]: could you put retry loop inside cachePop.Init() function and pass maxPopulatorRetries to it?

File: components/helm-broker/internal/ybundle/automock/extended.go [hunk 6]
```diff
@@ -22,33 +22,33 @@ func (m *Repository) ExpectErrorOnIndexReader(outErr error) {
 }
 
 func (m *Repository) ExpectOnBundleReader(givenName ybundle.BundleName, givenVersion ybundle.BundleVersion, bundleReader io.Reader) {
```
Comment (polskikiel) [inline components/helm-broker/internal/ybundle/automock/extended.go]: You can change the package name from `ybundle` to `bundle`.

File: components/helm-broker/internal/ybundle/populator.go [hunk 7]
```diff
@@ -57,31 +57,46 @@ func (b *Populator) Init() error {
...
-				return errors.Wrapf(err, "while reading bundle archive for name [%s] and version [%v]", entryName, v.Version)
+				failed = failed + 1
+				b.log.Warnf("Could not load bundle: %s", err.Error())
+			} else {
+				loaded = loaded + 1
 			}
-			defer bundleCloser()
+		}
+	}
+	b.log.Infof("Loading bundles completed. Successfully loaded %d bundles, failed %d bundles.", loaded, failed)
```
Comment (polskikiel) [inline components/helm-broker/internal/ybundle/populator.go]: Could you write a test which will check if we log this with a correct number of loaded/failed bundles?

File: components/helm-broker/internal/ybundle/populator_test.go [hunk 8]
```diff
@@ -38,7 +39,6 @@ func TestPopulatorInitHappyPath(t *testing.T) {
 	assert.NoError(t, err)
 
 	logSink.AssertLogged(t, logrus.InfoLevel, "Bundle with name [meme] and version [0.10.0] successfully stored")
-
 }
 
 func TestPopulatorErrorOnGetIndexFile(t *testing.T) {
```
Comment (polskikiel) [inline components/helm-broker/internal/ybundle/populator_test.go]: `TestPopulatorErrorOnGetIndexFile` but we expect `require.NoError(t, err)`
same for others

Comment (piotrmiskiewicz) [reply components/helm-broker/internal/ybundle/populator_test.go]: Test method names changed

File: components/helm-broker/internal/ybundle/automock/extended.go (hunk 6 above)
Comment (piotrmiskiewicz) [reply components/helm-broker/internal/ybundle/automock/extended.go]: changed

File: components/helm-broker/internal/ybundle/populator.go (hunk 7 above)
Comment (piotrmiskiewicz) [reply components/helm-broker/internal/ybundle/populator.go]: In my opinion, testing if some info log entry was done is not a good way, like it is described here: https://dzone.com/articles/never-test-logging

**Pull request merged**